src/datadog/metrics.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+// This component provides an interface, `Metric`, and specific classes for
+// Counter and Gauge metrics. A metric has a name, type, and set of key:value
+// tags associated with it. Metrics can be general to APM or language-specific.
+// General metrics have `common` set to `true`, and language-specific metrics
+// have `common` set to `false`.
+
+#include <atomic>
+#include <string>
+#include <vector>
+
+namespace datadog {
+namespace tracing {
+
+class Metric {
+  // The name of the metric that will be published. A transformation occurs
+  // based on the name and whether it is "common" or "language-specific" when it
+  // is recorded.
+  const std::string name_;
```

Comment:
`const`? _const_? Is this data member... immutable? _Is that why you used `const`?_

Still not local, so fair game.